crates/bevy_ecs/src/query/state.rs

```diff
@@ -249,6 +361,76 @@ where
         }
     }
 
+    /// Gets the read-only query results for the given [`World`] and array of [`Entity`], where the last change and
+    /// the current change tick are given.
+    pub(crate) fn get_multiple_read_only_manual<'s, 'w, const N: usize>(
+        &'s self,
+        world: &'w World,
+        entities: [Entity; N],
+        last_change_tick: u32,
+        change_tick: u32,
+    ) -> Result<[<Q::ReadOnlyFetch as Fetch<'w, 's>>::Item; N], QueryEntityError> {
+        // SAFE: fetch is read-only
+        let array_of_results = unsafe {
+            entities.map(|entity| {
+                self.get_unchecked_manual::<Q::ReadOnlyFetch>(
+                    world,
+                    entity,
+                    last_change_tick,
+                    change_tick,
+                )
+            })
+        };
+
+        // If any of the entities were not present, return an error
```

Comment:
Might be worth adding a TODO replace with rust-lang/rust#79711 once it's no longer feature-gated?

Reply:
```suggestion
        // TODO: Replace with TryMap once https://github.com/rust-lang/rust/issues/79711 is stabilized
        // If any of the entities were not present, return an error
```